src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,127 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            int updatedRows = statement.executeUpdate();
+            if (updatedRows < 1) {
+                throw new DataProcessingException("Expected to insert at "
+                        + "least to one row, but inserted 0 rows");
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't save a book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return getOptionalBook(resultSet,
+                        "Can't get data about book by id: " + id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't connect to DB "
+                    + "or SQL execute query is not correct", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> booksList = new ArrayList<>();
+        String sql = "SELECT * FROM books;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                booksList.add(getOptionalBook(resultSet,
+                        "Can't get data about book").orElseThrow());
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get any books", e);
+        }
+        return booksList;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            int updatedRows = statement.executeUpdate();
+            if (updatedRows < 1) {
```

Comment:
also, you try to update by id, so it means, that you have to affect exactly one row